### DISCUSSION TITLE
update 시 InnoDB 버퍼 풀에서 디스크로 데이터가 저장되는 시점?

### DISCUSSION THREAD

**Discussion by emost22** [Well-Founded-Confidence/RealMySQL#35] [Q&A]
![4 11 UPDATE 후 InnoDB 버퍼 풀과 데이터 파일 및 언두 영역의 변화](https://user-images.githubusercontent.com/60142959/231096950-2374086b-d0ed-40a0-8a70-145dce2e3bec.png)
update가 실행되면
1. InnoDB 버퍼 풀의 {PK, 변경되는 컬럼} 값이 언두 로그에 저장된다.
2. InnoDB 버퍼 풀의 데이터가 변경된다.
3. `InnoDB 버퍼 풀의 변경 내용이 디스크의 데이터 파일에 기록됐는지 여부는 시점에 따라 다를 수 있다.`??

3번 과정에서 저는 당연히 commit하는 시점에 디스크의 데이터가 변경되는 것으로 이해를 했는데,
다른 작업으로 인해 디스크의 데이터가 변경되는 일이 있을까요?

**Comment by Soyflat** [comment]
디스크의 데이터가 변경된다기보다, 디스크 io 횟수를 줄이기 위해서 시점에 따라 다르다고 한거 아닐까요? 추가적으로 상상이지만 여러 세션에서 하나의 로우값을 변경 시키면, 디스크에는 안쓰이고, 버퍼풀에서만 교체되는 경우도 있을까 궁금하네요!
